pkg/policy/api/v3/rule_validation.go

```diff
@@ -0,0 +1,325 @@
+// Copyright 2016-2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v3
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+)
+
+const (
+	maxPorts = 40
+	// MaxCIDRPrefixLengths is used to prevent compile failures at runtime.
+	MaxCIDRPrefixLengths = 40
+)
+
+type exists struct{}
+
+// Sanitize validates and sanitizes a policy rule. Minor edits such as
+// capitalization of the protocol name are automatically fixed up. More
+// fundamental violations will cause an error to be returned.
+func (r Rule) Sanitize() error {
+	for i := range r.Ingress {
+		if err := r.Ingress[i].sanitize(); err != nil {
+			return err
+		}
+	}
+
+	for i := range r.Egress {
+		if err := r.Egress[i].sanitize(); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (i *IngressRule) sanitize() error {
+	l3Members := map[string]bool{
```

Comment:
Add a comment specifying what exactly this represents, specifically that we do not support combinations of any of the contents of `L3Members` together.